persistence-modules/spring-data-cosmosdb/src/main/java/com/baeldung/spring/data/cosmosdb/controller/ProductController.java

```diff
@@ -46,8 +50,8 @@ public void delete(@PathVariable String id, @PathVariable String category) {
         return productService.findProductByName(name);
     }
 
-    @GetMapping(value = "/category/{category}")
-    public List<Product> getByCategory(@PathVariable String category) {
+    @GetMapping(value = "/category")
```

Comment:
you can remove this. in the end the user should just do:
.../products?category=bla and not products/category?category=bla

Reply:
I can't actually because there is another method in the controller getByName which will have same URi as this if we remove category.
Which causes ambiguity at startup time

Reply:
ok then we remove this method make the params not required either one or the other and you find out which one is set and use it